src/som/primitives/actors/PromisePrims.java

```diff
@@ -51,23 +60,46 @@ public boolean matches(final Object[] args, final ExpressionNode[] argNodes) {
              specializer = IsActorModule.class, noWrapper = true)
   public abstract static class CreatePromisePairPrim extends UnaryExpressionNode {
 
+    protected final AbstractBreakpointNode promiseResolverBreakpoint;
+    protected final AbstractBreakpointNode promiseResolutionBreakpoint;
+
     protected static final DirectCallNode create() {
       Dispatchable disp = SPromise.pairClass.getSOMClass().lookupMessage(
           withAndFactory, AccessModifier.PUBLIC);
       return Truffle.getRuntime().createDirectCallNode(((SInvokable) disp).getCallTarget());
     }
 
-    public CreatePromisePairPrim(final boolean eagerWrapper, final SourceSection source) { super(eagerWrapper, source); }
+    public CreatePromisePairPrim(final boolean eagerWrapper, final SourceSection source) {
+      super(eagerWrapper, source);
+
+      if (VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
+        Breakpoints breakpointCatalog = VM.getWebDebugger().getBreakpoints();
+        FullSourceCoordinate sourceCoord = SourceCoordinate.create(source);
+        this.promiseResolverBreakpoint   = insert(BreakpointNodeGen.create(breakpointCatalog.getPromiseResolverBreakpoint(sourceCoord)));
```

Comment:
since we have multiple places for this stuff, I wonder whether it wouldn't be better to have a helper method for the breakpoint node creation. A factory method, that internally checks the `VmSettings. TRUFFLE_DEBUGGER_ENABLED` flag.